### ISSUE TITLE
Duplicate locations reported for CS8618

### ISSUE THREAD

**Issue by bjornhellander** [dotnet/roslyn#79188]
I stumbled upon a weird looking thing when writing some analyzer tests. I accidentally got a CS8618 in a test and noticed that it had an additional location which was the same as the primary location:

```
Actual diagnostic:
// /0/Test0.cs(6,19): error CS8618: Non-nullable property 'TestProperty' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable.
DiagnosticResult.CompilerError("CS8618").WithSpan(6, 19, 6, 31).WithSpan(6, 19, 6, 31).WithArguments("property", "TestProperty"),
```

Notice the two identical calls to WithSpan. Is this intentional?

Seen in version 4.12.0

**Comment by RikkiGibson** [comment]
Thanks for reporting this issue. As far as I know, this is by design. CS8618 is really reported for a constructor, but, when the constructor is not explicitly declared, the primary location is on the property instead. Since it is convenient for diagnostic suppressors to always be able to know where the declaration of the property we are warning about is located, there is always an additional location which refers to the property, which it can check unconditionally.

**Comment by bjornhellander** [comment]
Ah, smart. Thank you!
